### PR TITLE
DAOS-2423 rdb: Use 63-bit Raft indices and terms

### PR DESCRIPTION
**Pull request by liw**
This patch updates the Raft submodule to pick up the 63-bit log indices
and terms. Other noteworthy changes include:

  - Remove dre_id from rdb_entry as it is only used by raft tests for
    debugging purposes.
  - Remove isi_padding from rdb_installsnapshot_in as it is no longer
    required by the new msg_installsnapshot_t.
  - Change "%d" to "%ld" for log indices and terms in log messages.
  - Fix some Doxygen and formatting issues.

Signed-off-by: Li Wei <wei.g.li@intel.com>

### CODE REVIEW

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Comments on needed changes are in daos-stack/raft#38.

**Comment by liw** [comment]
@brianjmurrell, the Ubuntu build failed like this:

> [2020-11-12T02:42:10.106Z] The following packages have unmet dependencies:
> [2020-11-12T02:42:10.106Z]  udev : Depends: libudev1 (= 245.4-4ubuntu3) but 245.4-4ubuntu3.3 is installed
> [2020-11-12T02:42:10.106Z]  zlib1g-dev : Depends: zlib1g (= 1:1.2.11.dfsg-2ubuntu1) but 1:1.2.11.dfsg-2ubuntu1.2 is installed
> [2020-11-12T02:42:10.106Z]  pbuilder-satisfydepends-dummy : Depends: libmercury-dev (= 2.0.0~rc1-1) but it is not going to be installed
> [2020-11-12T02:42:10.106Z]                                  Depends: libraft-dev (= 0.7.0) but it is not going to be installed
> [2020-11-12T02:42:10.106Z]  libgfortran5 : Depends: gcc-10-base (= 10-20200411-0ubuntu1) but 10.2.0-5ubuntu1~20.04 is installed
> [2020-11-12T02:42:10.106Z]  uuid-dev : Depends: libuuid1 (= 2.34-0.1ubuntu9) but 2.34-0.1ubuntu9.1 is installed
> [2020-11-12T02:42:10.106Z] Unable to resolve dependencies!  Giving up...

More packaging changes missing?

**Comment by brianjmurrell** [comment]
> @brianjmurrell, the Ubuntu build failed like this:
> 
> > [2020-11-12T02:42:10.106Z]                                  Depends: libraft-dev (= 0.7.0) but it is not going to be installed

This is what caused it to fail.  You specified an `=` for the version (i.e. rather than a `>=`), which is fine, but if you do that, you need to include the full _version_-_release_ in your specification.  Because that raft change is not tagged with 0.7.0, it's _release_ has a git-hash in it, which needs to be included in your dependent version specification.

File: debian/control [hunk 1]
```diff
@@ -29,7 +29,7 @@ Build-Depends: debhelper (>= 10),
                libboost-dev,
                libspdk-dev,
                libipmctl-dev,
-               libraft-dev,
+               libraft-dev (= 0.7.0),
```
Comment (brianjmurrell) [inline debian/control:32]:
```suggestion
               libraft-dev (= 0.7.0-1379.g3c6e9bd),
```

**Comment by liw** [comment]
> > @brianjmurrell, the Ubuntu build failed like this:
> > > [2020-11-12T02:42:10.106Z]                                  Depends: libraft-dev (= 0.7.0) but it is not going to be installed
> 
> This is what caused it to fail. You specified an `=` for the version (i.e. rather than a `>=`), which is fine, but if you do that, you need to include the full _version_-_release_ in your specification. Because that raft change is not tagged with 0.7.0, it's _release_ has a git-hash in it, which needs to be included in your dependent version specification.

I see. Should have figured that out myself. Added the full version; let's see how it goes this time.

**Comment by liw** [comment]
I'm afraid I still have done something wrong:

> 23:24:55  The following packages have unmet dependencies:
> 23:24:55   udev : Depends: libudev1 (= 245.4-4ubuntu3) but 245.4-4ubuntu3.3 is installed
> 23:24:55   zlib1g-dev : Depends: zlib1g (= 1:1.2.11.dfsg-2ubuntu1) but 1:1.2.11.dfsg-2ubuntu1.2 is installed
> 23:24:55   pbuilder-satisfydepends-dummy : Depends: libmercury-dev (= 2.0.0~rc1-1) but it is not going to be installed
> 23:24:55                                   Depends: libraft-dev (= 0.7.0-1379.g3c6e9bd) but it is not going to be installed
> 23:24:55   libgfortran5 : Depends: gcc-10-base (= 10-20200411-0ubuntu1) but 10.2.0-5ubuntu1~20.04 is installed
> 23:24:55   uuid-dev : Depends: libuuid1 (= 2.34-0.1ubuntu9) but 2.34-0.1ubuntu9.1 is installed
> 23:24:55  Unable to resolve dependencies!  Giving up...

The RPM builds also failed (so I changed daos.spec's raft-devel to full version as well):

> 23:24:39  Error: No Package found for raft-devel = 0.7.0

**Review by kccain — APPROVED** [review]

**Comment by liw** [comment]
> I'm afraid I still have done something wrong:
> 
> > 23:24:55  The following packages have unmet dependencies:
> > 23:24:55   udev : Depends: libudev1 (= 245.4-4ubuntu3) but 245.4-4ubuntu3.3 is installed
> > 23:24:55   zlib1g-dev : Depends: zlib1g (= 1:1.2.11.dfsg-2ubuntu1) but 1:1.2.11.dfsg-2ubuntu1.2 is installed
> > 23:24:55   pbuilder-satisfydepends-dummy : Depends: libmercury-dev (= 2.0.0~rc1-1) but it is not going to be installed
> > 23:24:55                                   Depends: libraft-dev (= 0.7.0-1379.g3c6e9bd) but it is not going to be installed
> > 23:24:55   libgfortran5 : Depends: gcc-10-base (= 10-20200411-0ubuntu1) but 10.2.0-5ubuntu1~20.04 is installed
> > 23:24:55   uuid-dev : Depends: libuuid1 (= 2.34-0.1ubuntu9) but 2.34-0.1ubuntu9.1 is installed
> > 23:24:55  Unable to resolve dependencies!  Giving up...
> 
> The RPM builds also failed (so I changed daos.spec's raft-devel to full version as well):
> 
> > 23:24:39  Error: No Package found for raft-devel = 0.7.0

@brianjmurrell, any more ideas on where to look?

**Comment by brianjmurrell** [comment]
@liw You need to add the `PR-repos: raft@PR-38` to every commit that you want to use that PR.  So your very last commit must have it for the build to use it.

**Review by kccain — DISMISSED** [review]

**Review by brianjmurrell — DISMISSED** [review]

File: debian/control [hunk 2]
```diff
@@ -29,7 +29,7 @@ Build-Depends: debhelper (>= 10),
                libboost-dev,
                libspdk-dev,
                libipmctl-dev,
-               libraft-dev,
+               libraft-dev (= 0.7.0-1377.g4faff0b),
```
Comment (kccain) [inline debian/control:32]: does this require such a specific value rather than 0.7.0?

**Review by kccain — APPROVED** [review]

File: debian/control (hunk 2 above)
Comment (brianjmurrell) [reply debian/control:32]: When it's an `=`, I believe it does.  If you look at c9b6493f42d1ec921aaac03787dc1a21d3cfb789 you can see @liw tried https://github.com/daos-stack/daos/blob/f6431fac9f31d8c38ebab6fbcac5322fcc136f2a/debian/control#L32 and in the [DEB Build](https://build.hpdd.intel.com/blue/rest/organizations/jenkins/pipelines/daos-stack/pipelines/daos/branches/PR-3826/runs/4/nodes/99/steps/310/log/?start=0) you can see that it couldn't find `libraft-dev (= 0.7.0)`.

If the `Build-Depends` were changed to `libraft-dev (>= 0.7.0)` that would probably work.
`

**Comment by liw** [comment]
CI failures
  - daos_test -V -DER_INVAL: DAOS-6022
  - osa offline parallel: DAOS-6107
  - osa online reintegration: DAOS-6123
  - mdtest small timeout: DAOS-6121
  - macsio dfuse mount: DAOS-6122
  - rebuild 22 timeout: DAOS-6124

**Review by kccain — APPROVED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**